File: merge-two-binary-trees/merge-two-binary-trees.py
class Solution:
    def mergeTrees(self, root1: TreeNode, root2: TreeNode) -> TreeNode:

        def merge(a: TreeNode, b: TreeNode) -> TreeNode:
            if a is None and b is None:
                return None
            
            if a and b:
                node = TreeNode(a.val + b.val)
                node.left = merge(a.left, b.left)
                node.right = merge(a.right, b.right)
            elif a and not b:
                node = TreeNode(a.val)
                node.left = merge(a.left, None)
                node.right = merge(a.right, None)
            elif not a and b:
                node = TreeNode(b.val)
                node.left = merge(None, b.left)
                node.right = merge(None, b.right)
            return node

        return merge(root1,root2)

File: merge-two-binary-trees/merge-two-binary-trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode mergeTrees(TreeNode root1, TreeNode root2) {
        return MergeTree(root1, root2);
    }

    private TreeNode MergeTree(TreeNode root1, TreeNode root2) {
        if (root1 == null & root2 == null) return null;

        TreeNode root = new TreeNode();
        root.val = (root1 != null? root1.val : 0) + (root2 != null? root2.val : 0);
        root.left = MergeTree(root1 != null? root1.left : null, root2 != null? root2.left : null);
        root.right = MergeTree(root1 != null? root1.right : null, root2 != null? root2.right : null);

        return root;
    }
}